rocAL/rocAL/include/parameters/parameter_crop.h

```diff
@@ -50,18 +50,17 @@ class CropParam
         x_drift_factor     = default_x_drift_factor();
         y_drift_factor     = default_y_drift_factor();
     }
-    void set_image_dimensions( const std::vector<uint32_t>& in_width_, const std::vector<uint32_t>& in_height_)
+    void set_image_dimensions(RocalROI *roi)
     {
-        if(in_width_.size() != in_width.size() || in_height.size() != in_height_.size())
-            THROW("wrong input width = "+ TOSTR(in_width.size())+" or height size = "+TOSTR(in_height_.size()))
-        in_width  = in_width_;
-        in_height = in_height_;
+        if(roi == nullptr)
```

Comment:
the roi is pointer to the struct passed, where as it is copied before. make sure this works in this case. Who owns the roi in this case?

Reply:
The ROI buffer is owned by the tensor and we are using the ptr here